core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -331,6 +331,30 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * A file name may contain some invalid url characters, such as " ". This method will convert the
+   * file name to a raw path accepted by `java.net.URI(String)`.
+   *
+   * Note: the file name must not contain "/" or "\"
+   */
+  def encodeFileNameToURIRawPath(fileName: String): String = {
```

Comment:
Yes, it must be `file:/abc:xyz` since you need to tell it it needs to be treated as a file name. Meh, at this point why not just `new URI("file:/" + fileName).getRawPath.substring(1)` and keep it simple?